kpops/manifests/strimzi/kafka_topic.py

```diff
@@ -0,0 +1,84 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from pydantic import ConfigDict, Field, model_validator
+from typing_extensions import Any
+
+from kpops.components.common.topic import KafkaTopic
+from kpops.manifests.kubernetes import KubernetesManifest, ObjectMeta
+from kpops.utils.docstring import describe_attr
+from kpops.utils.pydantic import CamelCaseConfigModel
+
+if TYPE_CHECKING:
+    try:
+        from typing import Self  # pyright: ignore[reportAttributeAccessIssue]
+    except ImportError:
+        from typing_extensions import Self
+
+
+class TopicSpec(CamelCaseConfigModel):
+    """Specification of a Kafka topic.
+
+    :param partitions: The number of partitions the topic should have. This cannot be decreased after topic creation. It can be increased after topic creation, but it is important to understand the consequences that has, especially for topics with semantic partitioning. When absent this will default to the broker configuration for `num.partitions`.
+    :param replicas: The number of replicas the topic should have. When absent this will default to the broker configuration for `default.replication.factor`.
+    :param config: The topic configuration.
+
+    """
+
+    partitions: int = Field(
+        default=1, ge=1, description=describe_attr("partitions", __doc__)
+    )
+    replicas: int = Field(
+        default=1, ge=1, le=32767, description=describe_attr("replicas", __doc__)
+    )
+    config: dict[str, str | int] = Field(
+        default_factory=dict, description=describe_attr("config", __doc__)
+    )
+
+    model_config = ConfigDict(extra="allow")
+
+    @model_validator(mode="before")
+    @classmethod
+    def set_defaults_if_none(cls, values: Any) -> Any:
+        if values.get("partitions") is None:
+            values["partitions"] = 1
+        if values.get("replicas") is None:
+            values["replicas"] = 1
+        return values
+
+
+class StrimziKafkaTopic(KubernetesManifest):
+    """Represents a Strimzi Kafka Topic CRD.
+
+    CRD definition: https://github.com/strimzi/strimzi-kafka-operator/blob/main/install/cluster-operator/043-Crd-kafkatopic.yaml
+    example: https://github.com/strimzi/strimzi-kafka-operator/blob/main/examples/topic/kafka-topic.yaml
+    """
+
+    api_version: str = "kafka.strimzi.io/v1beta2"
+    kind: str = "KafkaTopic"
+    metadata: ObjectMeta
+    spec: TopicSpec
+    status: dict[str, Any] | None = None
+
+    @classmethod
+    def create_strimzi_topic(cls, topic: KafkaTopic, bootstrap_servers: str) -> Self:
+        metadata = ObjectMeta.model_validate(
+            {
+                "name": topic.name,
+                "labels": {
+                    "strimzi.io/cluster": bootstrap_servers,
```

Comment:
are you sure about this? shouldn't it be the name of the Kafka cluster?

Reply:
That is correct. This is the documentation of the Topic Operator. The `strimzi.io/cluster` must match the env var `STRIMZI_RESOURCE_LABELS`.

https://strimzi.io/docs/operators/latest/deploying#deploying-the-topic-operator-standalone-str

Reply:
I still don't get it. according to the docs `STRIMZI_RESOURCE_LABELS` and `STRIMZI_KAFKA_BOOTSTRAP_SERVERS` are two independent settings

Reply:
ah, I didn't see that you changed it. looks good now